third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -549,6 +549,41 @@ void LoopPipeliner::emitPrologue() {
 }
 
 void LoopPipeliner::emitEpilogue(DenseMap<Value, Value> &newResults) {
+  // After pipelining, some of the depArgs have beem mapped to new args.
+  // We need to remap these.
+  //
+  // For example, if we have
+  //
+  //   ptr = ...
+  //   ret = scf.for iter_args(a_ptr=ptr)
+  //     a = load(a_ptr)
+  //     c += dot(a, ...)
+  //     a_ptr_new = a_ptr + N
+  //     scf.yield %a_ptr_new
+  //
+  // then the ptr arg should be mapped to a new arg in the for loop.
+  //
+  //   ptr = ...
+  //   load_pre = load(ptr)
+  //   ptr_new = ptr + N
+  //   ret = scf.for iter_args(a_ptr=ptr, ld=load_pre, A_ptr=ptr_new)
+  //     a_next = load(a_ptr)
+  //     c += dot(ld, ...)
+  //     A_ptr_new = A_ptr + N
+  //     scf.yield a_ptr, a_next, A_ptr_new
```

Comment:
There seems a typo in the code snippet. Does the following loop make sense?

  //   ptr = ...
  //   load_pre = load(ptr)
  //   ptr_new = ptr + N
  //   ret = scf.for iter_args(a_ptr=ptr_new, ld=load_pre)
  //     a_next = load(a_ptr)
  //     c += dot(ld, ...)
  //     A_ptr_new = a_ptr + N
  //     scf.yield A_ptr_new, a_next

Reply:
The code snippet you wrote is not what the current code generates. The current implementation adds new iter_arg variables to handle the updated args, which is reflected in the comment I added initially. I did try to add a bit more detail though to hopefully clarify.

Reply:
I see, but the `load(a_ptr)` should be `load(A_ptr)`?

Reply:
yes, you're right - thanks for catching that!